tests/test_qnode.py

```diff
@@ -26,135 +28,198 @@
+
+
+@pytest.fixture(scope="function")
+def mock_device():
+    """A mock instance of the abstract Device class"""
+
+    with patch.multiple(Device, __abstractmethods__=set()):
+        yield Device()
+
+
+@pytest.fixture(scope="function")
+def opqueue_test_node(mock_device):
+    """Provides a circuit for the subsequent tests of the operation queue"""
+
+    def circuit(x):
+        qml.RX(x, wires=[0])
+        qml.CNOT(wires=[0, 1])
+        qml.RY(0.4, wires=[0])
+        qml.RZ(-0.2, wires=[1])
+        return qml.expval(qml.PauliX(0)), qml.expval(qml.PauliZ(1))
+
+    node = qml.QNode(circuit, mock_device)
+    node.construct([1.0])
+
+    return node
 
 
+class TestQNodeOperationQueue:
+    """Tests that the QNode operation queue is properly filled and interacted with"""
+
+    def test_operation_ordering(self, opqueue_test_node):
+        """Tests that the ordering of the operations is correct"""
+
+        assert opqueue_test_node.ops[0].name == "RX"
+        assert opqueue_test_node.ops[1].name == "CNOT"
+        assert opqueue_test_node.ops[2].name == "RY"
+        assert opqueue_test_node.ops[3].name == "RZ"
+        assert opqueue_test_node.ops[4].name == "PauliX"
+        assert opqueue_test_node.ops[5].name == "PauliZ"
+
+    def test_op_successors_operations_only(self, opqueue_test_node):
+        """Tests that _op_successors properly extracts the successors that are operations"""
+
+        operation_successors = opqueue_test_node._op_successors(0, only="G")
+
+        assert opqueue_test_node.ops[0] not in operation_successors
+        assert opqueue_test_node.ops[1] in operation_successors
+        assert opqueue_test_node.ops[4] not in operation_successors
+
+    def test_op_successors_observables_only(self, opqueue_test_node):
+        """Tests that _op_successors properly extracts the successors that are observables"""
+
+        observable_successors = opqueue_test_node._op_successors(0, only="E")
+
+        assert opqueue_test_node.ops[0] not in observable_successors
+        assert opqueue_test_node.ops[1] not in observable_successors
+        assert opqueue_test_node.ops[4] in observable_successors
+
+    def test_op_successors_both_operations_and_observables(self, opqueue_test_node):
+        """Tests that _op_successors properly extracts all successors"""
+
+        successors = opqueue_test_node._op_successors(0, only=None)
+
+        assert opqueue_test_node.ops[0] not in successors
+        assert opqueue_test_node.ops[1] in successors
+        assert opqueue_test_node.ops[4] in successors
+
+    # TODO
+    # once _op_successors has been upgraded to return only strict successors using a DAG
+    # add a test that checks that the strict ordering is used
+    # successors = q._op_successors(2, only=None)
+    # self.assertTrue(q.ops[4] in successors)
+    # self.assertTrue(q.ops[5] not in successors)
+
+
+class BasicTest(BaseTest):
```

Comment:
I guess from here on its just a reformatting due to Black? But we should make a note here to refactor also these tests to pytest instead of unittest.

Reply:
Everything but the first routine that was already refactored to pytest is just black.